src/lib.rs

```diff
@@ -280,60 +314,120 @@ impl Log for SimpleLogger {
                 #[cfg(not(feature = "colored"))]
                 {
-                    record.level().to_string()
+                    Field::new(record.level().to_string(), LVL_FMT)
                 }
             };
 
             let target = if !record.target().is_empty() {
-                record.target()
+                Field::new(record.target(), TGT_FMT)
             } else {
-                record.module_path().unwrap_or_default()
+                Field::new(record.module_path().unwrap_or_default(), TGT_FMT)
             };
 
+            #[cfg(feature = "thread_ids")]
+            let thread_id = std::thread::current();
+
+            #[cfg(feature = "thread_ids")]
+            let target_thread = if self.threadids {
+                Field::new(thread_id.name().unwrap_or("UNKNOWN"), TRD_FMT)
+            } else {
+                Field::none()
+            };
+
+            #[cfg(not(feature = "thread_ids"))]
+            let target_thread = "";
+
             #[cfg(feature = "chrono")]
-            if self.timestamps {
-                #[cfg(not(feature = "stderr"))]
-                println!(
-                    "{} {:<5} [{}] {}",
-                    Local::now().format("%Y-%m-%d %H:%M:%S,%3f"),
-                    level_string,
-                    target,
-                    record.args()
-                );
-                #[cfg(feature = "stderr")]
-                eprintln!(
-                    "{} {:<5} [{}] {}",
-                    Local::now().format("%Y-%m-%d %H:%M:%S,%3f"),
-                    level_string,
-                    target,
-                    record.args()
-                );
-                return;
-            }
+            let timestamp = if self.timestamps {
+                Field::new(Local::now().format("%Y-%m-%d %H:%M:%S,%3f"), TS_FMT)
+            } else {
+                Field::none()
+            };
+
+            #[cfg(not(feature = "chrono"))]
+            let timestamp = Field::<char>::none();
 
             #[cfg(not(feature = "stderr"))]
-            println!("{:<5} [{}] {}", level_string, target, record.args());
+            println!(
+                LOGGING_FMT!(),
+                ts = timestamp,
+                lvl = level_string,
+                thr = target_thread,
+                tgt = target,
+                msg = record.args()
+            );
+
             #[cfg(feature = "stderr")]
-            eprintln!("{:<5} [{}] {}", level_string, target, record.args());
+            eprintln!(
+                LOGGING_FMT!(),
+                ts = timestamp,
+                lvl = level_string,
+                thr = target_thread,
+                tgt = target,
+                msg = record.args()
+            );
         }
     }
 
     fn flush(&self) {}
 }
 
+// implementation detail:
+//  logged fields can be optional and are suppressed when 'None'
+//  For formatting purposes each field has a prefix and a suffix which are
+//  printed around it
+struct Field<T> {
```

Comment:
I think it would be useful to define this before `SimpleLogger` so the file is easier to read top-to-bottom.